docs/docs/02-concepts/01-preflight-and-inflight.md

```diff
@@ -1,230 +1,192 @@
 
-## Preflight
+Every application built to run on the cloud can be thought of as being made up of two kinds of components.
 
-TODO
+The first set of components are the resources and network-accessible services that your application uses.
+These resources are typically optimized for use cases like storing data or files (like buckets), processing HTTP requests (like API gateways), distributing content (like CDNs), or running user-defined code (such as Lambda functions or Kubernetes deployments).
 
-## Inflight 
+The second set of components are the pieces of user-defined code that you write to interact with these resources.
+For example, you may have code runs in a Lambda function or a Kubernetes cluster that makes requests to the internet, perform queries on your database, transforms data from storage, and does any number of other operations in response to an event.
 
-Inflights (or "inflight functions") are Wing's distributed computing primitive.
-They are isolated code blocks which can be packaged and executed on compute
-platforms in the cloud (such as containers, CI/CD pipelines or FaaS).
+Wing lets you express these two parts of a cloud application through the concepts of **preflight code** and **inflight code**.
 
-Inflights can seamlessly interact with resources through the resource's inflight
-API.
+## Preflight code
 
-The following example shows an inflight which puts an object inside a bucket,
-hosted in a cloud function:
+Preflight is where you define your app's infrastructure.
+The entrypoint, or default phase, of every Wing program is the preflight phase.
 
-```js
+For example, this code defines a storage bucket, and populates the bucket with a piece of static data when the app is deployed.
+
+```js playground
 bring cloud;
 
-let bucket = new cloud.Bucket();
+let data = new cloud.Bucket();
 
-new cloud.Function(inflight (_: str): str => {
-  bucket.put("hello.txt", "world");
-});
+let orders = { "hummus": 100 };
+data.addObject("orders.json", Json.stringify(orders));
 ```
 
-But before we explain what inflight functions are (we also call them
-"inflights"), let's take a step back and talk about what are cloud applications
-anyway?
+`Bucket` is a class, and `addObject()` is a preflight method of `Bucket`.
 
-A cloud application is software that deeply leverages **cloud resources** in
-order to perform its duties. A cloud resource is *any network-accessible
-service*. These resources can be used to store data or files (like buckets),
-process HTTP requests (like API gateways), distribute content (like CDNs),
-publish notifications to mobile phones or run user-defined code (such as Lambda
-functions or Kubernetes deployments). Cloud providers such as AWS, GCP and Azure
-offer a basic set of resources (similar to how an operating system offers a
-basic set of resources to applications running on the machine), but that doesn't
-imply that those are the only resources an application might use.
+Preflight code is executed at compile-time.
```

Comment:
To someone new to wing, my guess is that "executed at compile-time" may require expanding on in the context of things like `new cloud.Bucket()` rather than just regular math and logging.

Reply:
Got it - I've added some more detail to try and clarify this area better